### DISCUSSION TITLE
Will loading multiple tailwind CSS builds from the same config cause issues?

### DISCUSSION THREAD

**Discussion by johnathanludwig** [tailwindlabs/tailwindcss#7227] [Help] [answered]
tldr; Are there any major issues with loading multiple tailwind builds that include a mix of the same and different classes sourced from the same tailwind config on one page? If so, what variants, etc, would be the main concern?

I'm working on a large application that is broken down into multiple small pieces that come together at runtime. There is a main application that has some UI elements like navigation and brings together other smaller applications(groups of pages) at runtime. Both the main app and micro apps also use a shared package library that is built using tailwind. Currently there is one tailwind v2 CSS file being included by the main app, but isn't purged because we don't know what classes the smaller apps will use as they are built/deployed separately. We've limited the variants and such to save space, but it's still ~800kb. 😬

I'd like to upgrade to tailwind v3 with JIT, and have the smaller apps also include their own tailwind v3 with JIT, so they could all build the CSS they need. They would use the same tailwind config, so colors, spacing, etc would be the same between them. Probably only have the primary one do the reset?

My assumption is that if they have the same config it seems like the config the classes would always generate the same so there shouldn't be a concern with two builds defining the same class. There would be some duplication, but definitely less overall than what we currently have, and with v3 we would get the flexibility of having all the variants available. 

So if my assumptions are correct, the only issue I can potentially foresee becomes a second CSS file defining similar classes and breaking some CSS loaded by the first one. Are there any specific variants, etc, that might be an issue? Anything else we should lookout for?

An alternative would be to prefix classes in each micro app, but I'd really like to avoid that if possible. Also, with the shared package library between them there will still be some CSS that would overlap without prefixes.

**Comment by johnathanludwig** [comment, marked as answer]
So as a follow up. We've been doing this for a few months and it's been fine. 

We have had a few instances where the order of CSS generated was slightly different in some apps based on the usage, but they were only noticed because two competing classes existed on the same elements.

One thing we made sure to use was [Webpack's lazy styles](https://webpack.js.org/loaders/style-loader/#lazystyletag), so the CSS only gets mounted when a module is loaded, and when it get's removed the styles it provided get removed as well.
